ci/scripts/build_kernel_loader.sh

```diff
@@ -43,6 +45,20 @@ mv arrow_tmp/arrow-${sha}/cpp/src/arrow/chunk_resolver.cc arrow_vendored/cpp/src
 mv arrow_tmp/arrow-${sha}/cpp/src/arrow/chunk_resolver.h arrow_vendored/cpp/src/arrow/
 mv arrow_tmp/arrow-${sha}/cpp/src/arrow/compute/kernels/codegen_internal.h arrow_vendored/cpp/src/arrow/compute/kernels/
 mv arrow_tmp/arrow-${sha}/cpp/src/arrow/compute/kernels/vector_rank.cc arrow_vendored/cpp/src/arrow/compute/kernels/
+# Pivot Kernel requirements
+mv arrow_tmp/arrow-${sha}/cpp/src/arrow/compute/api_aggregate.cc arrow_vendored/cpp/src/arrow/compute/
+mv arrow_tmp/arrow-${sha}/cpp/src/arrow/compute/api_aggregate.h arrow_vendored/cpp/src/arrow/compute/
+mv arrow_tmp/arrow-${sha}/cpp/src/arrow/compute/kernels/aggregate_pivot.cc arrow_vendored/cpp/src/arrow/compute/kernels/
+mv arrow_tmp/arrow-${sha}/cpp/src/arrow/compute/kernels/hash_aggregate.cc arrow_vendored/cpp/src/arrow/compute/kernels/
+mv arrow_tmp/arrow-${sha}/cpp/src/arrow/compute/kernels/aggregate_internal.h arrow_vendored/cpp/src/arrow/compute/kernels/
+mv arrow_tmp/arrow-${sha}/cpp/src/arrow/compute/kernels/util_internal.h arrow_vendored/cpp/src/arrow/compute/kernels/
+mv arrow_tmp/arrow-${sha}/cpp/src/arrow/util/int128_internal.h arrow_vendored/cpp/src/arrow/util/
+mv arrow_tmp/arrow-${sha}/cpp/src/arrow/compute/kernels/common_internal.h arrow_vendored/cpp/src/arrow/compute/kernels/
+mv arrow_tmp/arrow-${sha}/cpp/src/arrow/compute/kernels/pivot_internal.cc arrow_vendored/cpp/src/arrow/compute/kernels/
+mv arrow_tmp/arrow-${sha}/cpp/src/arrow/compute/kernels/pivot_internal.h arrow_vendored/cpp/src/arrow/compute/kernels/
+# Specific for hash_pivot
+mv arrow_tmp/arrow-${sha}/cpp/src/arrow/compute/kernels/aggregate_var_std_internal.h arrow_vendored/cpp/src/arrow/compute/kernels/
```

Comment:
Looks like we don't vendor namespaces for all files - how are we sure these don't have symbol conflicts?

Reply:
@pitrou this is my understanding, do you think that's correct about not requiring to update the namespace of the internal header files?

Reply:
> We could be more aggressive to avoid those issues and build more internal arrow things on the kernel-loader 

That would probably be safer indeed. The DLL would be bigger, but I'm not sure that's a problem.

Reply:
I don't think DLL size is a concern - my main concern is just the mount of work to add kernels and maintain this build.

@raulcd 

`function_internal.cc` and `codegen_internal.cc` seems pretty light weight
`aggregate_internal.cc` looks like it is under Acero so we shouldn't need? (kernels dont depend on Acero)

What else do you think we need to vendor?

Reply:
I am going to give that a try. I initially tried and it required me to vendor more and more things because some of those internal implementation source files require others, etcetera. I'll probably merge this PR and create a new one on top to keep the scope of the PRs contained. @icexelloss I can create a tag with quantile rank and rank normal kernels only, current main, if you want.

Reply:
(Not saying we should do it - just trying to understand the how much extra work is needed for the safety)

Reply:
> `aggregate_internal.cc` looks like it is under Acero so we shouldn't need? (kernels dont depend on Acero)
Correct, I saw an `aggregate_internal.cc` and assumed incorrectly.

From the internal headers we pull I've identified:
```
function_internal.cc
codegen_internal.cc
util_internal.cc
aggregate_var_std_internal.cc
row_encoder_internal.cc
```
Updating those might require us to apply some diffs to some of the other headers as some of the symbols will be on a different namespace.

Reply:
> @icexelloss I can create a tag with quantile rank and rank normal kernels only, current main, if you want.

No need - let's just do it in one go

>I'll probably merge this PR and create a new one on top to keep the scope of the PRs contained.

Sounds good - I already tested this internal and it seems fine.